storage/connectors/bigquery/src/main/java/feast/storage/connectors/bigquery/statistics/FieldStatisticsQueryInfo.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.storage.connectors.bigquery.statistics;
+
+import feast.core.FeatureSetProto.EntitySpec;
+import feast.core.FeatureSetProto.FeatureSpec;
+import feast.types.ValueProto.ValueType.Enum;
+
+/**
+ * Value class for Features containing information necessary to template stats-retrieving queries.
+ */
+public class FieldStatisticsQueryInfo {
```

Comment:
Just my own curiousity. How are we deciding when to use AutoValue and when to not use AutoValue?

In this case we have a value class which seems like it fits the AutoValue use case well, but we opt not to use it. Then for complex classes like `BigQueryStatisticsRetriever` we do opt for AutoValue.

Reply:
I personally use autovalue when there are classes that require more than 2 elements to construct, but pebble is unfortunately incompatible with autovalue classes, so in this case I can't.